Bonsai.Design/ObjectTextVisualizer.cs

```diff
@@ -30,25 +31,99 @@ public class ObjectTextVisualizer : BufferedVisualizer
         /// <inheritdoc/>
         protected override void ShowBuffer(IList<Timestamped<object>> values)
         {
-            if (values.Count > 0)
+            if (values.Count == 0)
+                return;
+
+            var sb = new StringBuilder();
+
+            // Trim old values if bufferSize was reduced
+            while (buffer.Count >= bufferSize)
+                buffer.Dequeue();
+
+            // Add new values to the buffer (and only the ones which might appear)
+            for (int i = Math.Max(0, values.Count - bufferSize); i < values.Count; i++)
             {
-                base.ShowBuffer(values);
-                textBox.Text = string.Join(Environment.NewLine, buffer);
-                textPanel.Invalidate();
+                sb.Clear();
+                AppendDisplayText(sb, values[i]);
```

Comment:
```suggestion
                AppendDisplayText(sb, values[i].Value);
```

To keep the current default value we need to get the value here to avoid printing the timestamp.

Reply:
Oops, that's what I get for changing my test workflow. (I added a timer and thought the timestamps were what it was emitting, didn't notice they were on the other lines too.)